internal/services/manager_test.go

```diff
@@ -0,0 +1,90 @@
+package services_test
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/services"
+	"github.com/ubuntu/authd/internal/testutils"
+	"google.golang.org/grpc"
+)
+
+func TestNewManager(t *testing.T) {
+	tests := map[string]struct {
+		cacheDir string
+
+		systemBusSocket string
+
+		wantErr bool
+	}{
+		"Successfully create the manager": {},
+
+		"Error when can not create cache":          {cacheDir: "doesnotexist", wantErr: true},
+		"Error when can not create broker manager": {systemBusSocket: "doesnotexist", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			if tc.cacheDir == "" {
+				tc.cacheDir = t.TempDir()
+			}
+			if tc.systemBusSocket != "" {
+				t.Setenv("DBUS_SYSTEM_BUS_ADDRESS", tc.systemBusSocket)
+			}
+
+			_, err := services.NewManager(context.Background(), tc.cacheDir, nil)
+			if tc.wantErr {
+				require.Error(t, err, "NewManager should have returned an error, but did not")
+				return
+			}
+			require.NoError(t, err, "NewManager should not have returned an error, but did")
```

Comment:
shouldn’t you defer m.Stop() to not leak anything? (even if there is a test for that, but I’m unsure this is a real test btw, as we don’t have error cases for `stop()` tested, so it could be in New() as "this is the lifecycle of a manager?)

Reply:
What? Doesn't the connection close itself after the lifetime ends?
Sorry, too much Rust recently... :smiling_face_with_tear: